tree.go

```diff
@@ -1542,14 +1542,14 @@ func (n *InternalNode) collectNonHashedNodes(list []VerkleNode, paths [][]byte,
 	list = append(list, n)
 	paths = append(paths, path)
 	for i, child := range n.children {
-		childpath := make([]byte, len(path)+1)
-		copy(childpath, path)
-		childpath[len(path)] = byte(i)
```

Comment:
This was unfortunately creating the L1545 slice for every 256 children on every `InternalNode`, even if those children's where empty (or any other case not used in the switch below).